src/Catalyst.Core.Modules.KeySigner/KeySigner.cs

```diff
@@ -22,71 +22,51 @@
 using Catalyst.Abstractions.Types;
 using Catalyst.Protocol;
 using Catalyst.Protocol.Cryptography;
-using Catalyst.Protocol.Network;
-using Org.BouncyCastle.Security;
+using Org.BouncyCastle.Crypto.Parameters;
 
 namespace Catalyst.Core.Modules.KeySigner
 {
     public sealed class KeySigner : IKeySigner
     {
-        private readonly IKeyStore _keyStore;
         private readonly ICryptoContext _cryptoContext;
+        private readonly IKeyApi _keyApi;
         private readonly IKeyRegistry _keyRegistry;
         private readonly KeyRegistryTypes _defaultKey = KeyRegistryTypes.DefaultKey;
 
         /// <summary>Initializes a new instance of the <see cref="KeySigner"/> class.</summary>
         /// <param name="keyStore">The key store.</param>
         /// <param name="cryptoContext">The crypto context.</param>
         /// /// <param name="keyRegistry">The key registry.</param>
-        public KeySigner(IKeyStore keyStore, 
-            ICryptoContext cryptoContext, 
+        public KeySigner(ICryptoContext cryptoContext,
+            IKeyApi keyApi,
             IKeyRegistry keyRegistry)
         {
-            _keyStore = keyStore;
             _cryptoContext = cryptoContext;
+            _keyApi = keyApi;
             _keyRegistry = keyRegistry;
-            InitialiseKeyRegistry();
         }
 
-        private void InitialiseKeyRegistry()
-        {
-            if (!TryPopulateDefaultKeyFromKeyStore(out _))
-            {
-                GenerateKeyAndPopulateRegistryWithDefaultAsync().ConfigureAwait(false);
-            }   
-        }
-
-        private async Task GenerateKeyAndPopulateRegistryWithDefaultAsync()
-        {
-            var privateKey = await _keyStore.KeyStoreGenerateAsync(NetworkType.Devnet, _defaultKey).ConfigureAwait(false);
-            if (privateKey != null)
-            { 
-                _keyRegistry.AddItemToRegistry(_defaultKey, privateKey);
-            }
-        }
-
-        /// <inheritdoc/>
-        IKeyStore IKeySigner.KeyStore => _keyStore;
-
         /// <inheritdoc/>
         ICryptoContext IKeySigner.CryptoContext => _cryptoContext;
 
-        private IPrivateKey GetPrivateKey(KeyRegistryTypes keyIdentifier)
+        public IPublicKey GetPublicKey(KeyRegistryTypes keyIdentifier)
         {
-            var privateKey = _keyRegistry.GetItemFromRegistry(keyIdentifier);
-            if (privateKey == null && !TryPopulateRegistryFromKeyStore(keyIdentifier, out privateKey))
-            {
-                throw new SignatureException("The signature cannot be created because the key does not exist");
-            }
+            var publicKey = (Ed25519PublicKeyParameters) _keyApi.GetPublicKeyAsync("self").GetAwaiter().GetResult();
+            var publicKeyBytes = publicKey.GetEncoded();
+            return _cryptoContext.GetPublicKeyFromBytes(publicKeyBytes);
+        }
 
-            return privateKey;
+        public IPrivateKey GetPrivateKey(KeyRegistryTypes keyIdentifier)
```

Comment:
@Xela101 So are we not going to be using the KeyRegistry anymore to cache keys while the node is running?

Reply:
Oops, yeah I'll add the key registry back. I thought DFS used it's own underling key registry but looks like the key name is just mapped to the filesystem. Cheers.